src/main/kotlin/com/novatecgmbh/eventsourcing/axon/project/task/query/TaskProjector.kt

```diff
@@ -0,0 +1,97 @@
+package com.novatecgmbh.eventsourcing.axon.project.task.query
+
+import TaskCompletedEvent
+import TaskCreatedEvent
+import TaskDescriptionUpdatedEvent
+import TaskRescheduledEvent
+import TaskStartedEvent
+import com.novatecgmbh.eventsourcing.axon.project.task.api.TaskQuery
+import com.novatecgmbh.eventsourcing.axon.project.task.api.TasksByProjectQuery
+import com.novatecgmbh.eventsourcing.axon.project.task.command.TaskId
+import com.novatecgmbh.eventsourcing.axon.project.task.command.TaskStatusEnum.*
+import java.util.*
+import org.axonframework.config.ProcessingGroup
+import org.axonframework.eventhandling.EventHandler
+import org.axonframework.eventhandling.ResetHandler
+import org.axonframework.eventhandling.SequenceNumber
+import org.axonframework.extensions.kotlin.emit
+import org.axonframework.queryhandling.QueryHandler
+import org.axonframework.queryhandling.QueryUpdateEmitter
+import org.springframework.stereotype.Component
+
+@Component
+@ProcessingGroup("task-projector")
+class TaskProjector(
+    private val repository: TaskProjectionRepository,
+    private val queryUpdateEmitter: QueryUpdateEmitter
+) {
+  @EventHandler
+  fun on(event: TaskCreatedEvent, @SequenceNumber aggregateVersion: Long) {
+    saveProjection(
+        TaskProjection(
+            identifier = event.identifier,
+            version = aggregateVersion,
+            projectId = event.projectId,
+            name = event.name,
+            description = event.description,
+            startDate = event.startDate,
+            endDate = event.endDate,
+            status = PLANNED))
+  }
+
+  @EventHandler
+  fun on(event: TaskDescriptionUpdatedEvent, @SequenceNumber aggregateVersion: Long) {
+    updateProjection(event.identifier) {
+      it.name = event.name
+      it.description = event.description
+    }
+  }
+
+  @EventHandler
+  fun on(event: TaskRescheduledEvent, @SequenceNumber aggregateVersion: Long) {
+    updateProjection(event.identifier) {
+      it.startDate = event.startDate
+      it.endDate = event.endDate
+    }
+  }
+
+  @EventHandler
+  fun on(event: TaskStartedEvent, @SequenceNumber aggregateVersion: Long) {
+    updateProjection(event.identifier) { it.status = STARTED }
+  }
+
+  @EventHandler
+  fun on(event: TaskCompletedEvent, @SequenceNumber aggregateVersion: Long) {
+    updateProjection(event.identifier) { it.status = COMPLETED }
+  }
+
+  private fun updateProjection(identifier: TaskId, stateChanges: (TaskProjection) -> Unit) {
+    repository.findById(identifier).get().also {
```

Comment:
can fail if task with identifier is not found. is this intended?

Reply:
I suggest we leave it for now since this would be a severe error anyways.